substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/SubstrateExitHandlerFeature.java

```diff
@@ -34,7 +34,7 @@
 public class SubstrateExitHandlerFeature implements InternalFeature {
     @Override
     public void beforeAnalysis(BeforeAnalysisAccess access) {
-        if (SubstrateOptions.InstallExitHandlers.getValue()) {
+        if (SubstrateOptions.InstallExitHandlers.getValue() || VMInspectionOptions.hasJfrSupport()) {
```

Comment:
I assume `|| VMInspectionOptions.hasNativeMemoryTrackingSupport()` is intentionally missing due to not being present in graalvm 23.1 (for JDK 21)

Reply:
Yes, that's right, the native memory tracking feature isn't available yet in graalVM for JDK21.